Add Power == watts base unit to naming.md

mhansen opened the pull request:
I'm just making an IoT power device and I thought this might be nice to standardize in the docs.

brian-brazil commented:
Thanks for your PR. The preferred way to export this is joules, as when you have a counter in joules and take a rate on it you end up with watts - but the underlying unit in the metric names should still be the joule.

mhansen commented:
Great point. Watts only really makes sense for instantaneous gauges. How
about I amend this to spell out that you should probably just use joules?

On Sat, 27 Jun 2020 at 13:23, Brian Brazil <notifications@github.com> wrote:

> Thanks for your PR. The preferred way to export this is joules, as when
> you have a counter in joules and take a rate on it you end up with watts -
> but the underlying unit in the metric names should still be the joule.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/docs/pull/1687#issuecomment-650480782>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZYOJH7NI4IJ6AHBL3JG3RYVQ2TANCNFSM4OJZ6ARQ>
> .
>

mhansen commented:
Just writing this up, realised perhaps amperes should be rate(coulombs) by
the same logic?

On Sat, 27 Jun 2020 at 13:28, Mark Hansen <mark@markhansen.co.nz> wrote:

> Great point. Watts only really makes sense for instantaneous gauges. How
> about I amend this to spell out that you should probably just use joules?
>
> On Sat, 27 Jun 2020 at 13:23, Brian Brazil <notifications@github.com>
> wrote:
>
>> Thanks for your PR. The preferred way to export this is joules, as when
>> you have a counter in joules and take a rate on it you end up with watts -
>> but the underlying unit in the metric names should still be the joule.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/prometheus/docs/pull/1687#issuecomment-650480782>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAAZYOJH7NI4IJ6AHBL3JG3RYVQ2TANCNFSM4OJZ6ARQ>
>> .
>>
>

mhansen commented:
I've updated it to provide your guidance for power, WDYT? (I'm OK to drop
this if this is over-the-top documentation, or perhaps not in keeping with
the rest of this section).

On Sat, 27 Jun 2020 at 16:39, Mark Hansen <mark@markhansen.co.nz> wrote:

> Just writing this up, realised perhaps amperes should be rate(coulombs) by
> the same logic?
>
> On Sat, 27 Jun 2020 at 13:28, Mark Hansen <mark@markhansen.co.nz> wrote:
>
>> Great point. Watts only really makes sense for instantaneous gauges. How
>> about I amend this to spell out that you should probably just use joules?
>>
>> On Sat, 27 Jun 2020 at 13:23, Brian Brazil <notifications@github.com>
>> wrote:
>>
>>> Thanks for your PR. The preferred way to export this is joules, as when
>>> you have a counter in joules and take a rate on it you end up with watts -
>>> but the underlying unit in the metric names should still be the joule.
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/prometheus/docs/pull/1687#issuecomment-650480782>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AAAZYOJH7NI4IJ6AHBL3JG3RYVQ2TANCNFSM4OJZ6ARQ>
>>> .
>>>
>>

brian-brazil commented on content/docs/practices/naming.md:
```diff
@@ -84,4 +84,4 @@ The list is not exhaustive.
 | Energy | joules | |
 | Mass   | grams | _grams_ is preferred over _kilograms_ to avoid issues with the _kilo_ prefix. |
- 
+| Power  | | Prefer exporting a counter of joules, then `rate(joules)` gives you power in Watts.  |
```
Let's use rate(joules[5m]) so the syntax is valid.

brian-brazil commented:
> Perhaps amperes should be rate(coulombs) by the same logic?

That's be consistent, however I don't think I've come across a device that ever exposes that. Whereas both the joule and watt variants exist. I'd not generally try to cover everything, bytes and seconds are the main ones and the examples will hopefully provide enough reasoning so that developers end up largely doing something sane.

mhansen commented:
Yeah, I’ve also never found something exposing raw coulombs either. Sgtm.
I’ll adapt the syntax soon.

On Sat, 27 Jun 2020 at 19:41, Brian Brazil <notifications@github.com> wrote:

> Perhaps amperes should be rate(coulombs) by the same logic?
>
> That's be consistent, however I don't think I've come across a device that
> ever exposes that. Whereas both the joule and watt variants exist. I'd not
> generally try to cover everything, bytes and seconds are the main ones and
> the examples will hopefully provide enough reasoning so that developers end
> up largely doing something sane.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/docs/pull/1687#issuecomment-650532311>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZYOPRZ44Q64N5RKHIGGLRYW5EXANCNFSM4OJZ6ARQ>
> .
>

brian-brazil commented on content/docs/practices/naming.md:
```diff
@@ -84,4 +84,4 @@ The list is not exhaustive.
 | Energy | joules | |
 | Mass   | grams | _grams_ is preferred over _kilograms_ to avoid issues with the _kilo_ prefix. |
- 
+| Power  | watts | Prefer exporting a counter of joules, then `rate(joules[5m])` gives you power in Watts. |
```
You added a base unit since the last push, watts still aren't the base unit. If we were to put anything in here it would be joules.

Also this should go beside the energy entry - like we do with bits&bytes.

mhansen commented:
Moved besides energy, and removed the unit (sorry, thought it might be
useful to say watts is the extension but prefer joules).

On Sat, 27 Jun 2020 at 21:39, Brian Brazil <notifications@github.com> wrote:

> *@brian-brazil* commented on this pull request.
> ------------------------------
>
> In content/docs/practices/naming.md
> <https://github.com/prometheus/docs/pull/1687#discussion_r446515994>:
>
> > @@ -84,4 +84,4 @@ The list is not exhaustive.
>  | Electric current | amperes | |
>  | Energy | joules | |
>  | Mass   | grams | _grams_ is preferred over _kilograms_ to avoid issues with the _kilo_ prefix. |
> -
> +| Power  | watts | Prefer exporting a counter of joules, then `rate(joules[5m])` gives you power in Watts. |
>
> You added a base unit since the last push, watts still aren't the base
> unit. If we were to put anything in here it would be joules.
>
> Also this should go beside the energy entry - like we do with bits&bytes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/docs/pull/1687#pullrequestreview-438691819>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZYOKHNPOY5MMA2MRPIYLRYXLAHANCNFSM4OJZ6ARQ>
> .
>

brian-brazil commented:
Thanks!

brian-brazil commented:
> sorry, thought it might be useful to say watts is the extension but prefer joules

It's a bit of a grey area. Things get a little messy once you move beyond bytes and seconds.

mhansen commented:
Thanks for your help on this, I appreciate the guidance and I’m going to
miss a lot less detail exporting joules :-)

On Sat, 27 Jun 2020 at 21:56, Brian Brazil <notifications@github.com> wrote:

> sorry, thought it might be useful to say watts is the extension but prefer
> joules
>
> It's a bit of a grey area. Things get a little messy once you move beyond
> bytes and seconds.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/docs/pull/1687#issuecomment-650550971>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAZYOKFQOYUIDUPYAKT4MLRYXNAJANCNFSM4OJZ6ARQ>
> .
>
